tests/foreman/ui/test_host.py

```diff
@@ -2404,3 +2405,162 @@ def test_all_hosts_bulk_delete(target_sat, function_org, function_location, new_
+
+    :expectedresults: Job invocation page should be correctly generated
+        by the change content source action, generated script should also be correct
+
+    :CaseComponent:Hosts-Content
+
+    :Team: Phoenix-content
+    """
+
+    module_target_sat, org, lce, capsule, content_view, loc, ak = change_content_source_prep
+
+    rhel_contenthost.register(org, loc, ak.name, module_target_sat)
+
+    with module_target_sat.ui_session() as session:
+        session.organization.select(org_name=org.name)
+        session.location.select(loc_name=ANY_CONTEXT['location'])
+
+        # STEP 1: Test the part where you use "Update hosts manually" button
+        # Set the content source to the checked-out capsule
+        # Check that generated script contains correct name of new content source
+        rhel_contenthost_pre_values = rhel_contenthost.nailgun_host.content_facet_attributes
+        generated_script = session.host.change_content_source_get_script(
+            entities_list=[
+                rhel_contenthost.hostname,
+            ],
+            content_source=capsule[0]['name'],
+            lce=lce.name,
+            content_view=content_view.name,
+        )
+        rhel_contenthost_post_values = rhel_contenthost.nailgun_host.content_facet_attributes
+        content_source_from_script = re.search(r'--server.hostname=\"(.*?)\"', generated_script)
+
+        assert content_source_from_script.group(1) == capsule[0]['name']
+        assert rhel_contenthost_post_values['content_source']['name'] == capsule[0]['name']
+        assert rhel_contenthost_post_values['content_view']['name'] == content_view.name
+        assert rhel_contenthost_post_values['lifecycle_environment']['name'] == lce.name
+
+        session.browser.refresh()
+
+        # Step 2: Test the part where you use "Run job invocation" button
+        # Change the rhel_contenthost's content source back to what it was before STEP 1
+        # Check the prefilled job invocation page
+        session.host.change_content_source(
+            entities_list=[
+                rhel_contenthost.hostname,
+            ],
+            content_source=rhel_contenthost_pre_values['content_source']['name'],
+            lce=rhel_contenthost_pre_values['lifecycle_environment']['name'],
+            content_view=rhel_contenthost_pre_values['content_view']['name'],
+            run_job_invocation=True,
+        )
+        # Getting the data from the prefilled job invocation form
+        selected_category_and_template = session.jobinvocation.get_job_category_and_template()
+        selected_targeted_hosts = session.jobinvocation.get_targeted_hosts()
+
+        assert selected_category_and_template['job_category'] == 'Katello'
+        assert (
+            selected_category_and_template['job_template']
+            == 'Configure host for new content source'
+        )
+        assert selected_targeted_hosts['selected_hosts'] == [rhel_contenthost.hostname]
```

Comment:
Maybe we should check also from the host that the repo can be pulled from / is available